map/autoware_lanelet2_map_validator/src/validators/crosswalk/missing_regulatory_elements_for_crosswalk.cpp

```diff
@@ -0,0 +1,104 @@
+// Copyright 2024 Autoware Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "lib/utils.hpp"
+#include "validators/crosswalk/missing_regulatory_elements_for_crosswalks.hpp"
+
+#include <range/v3/view/filter.hpp>
+
+#include <lanelet2_core/LaneletMap.h>
+
+#include <set>
+
+namespace lanelet
+{
+namespace validation
+{
+namespace
+{
+lanelet::validation::RegisterMapValidator<MissingRegulatoryElementsForCrosswalksValidator> reg;
+}
+
+lanelet::validation::Issues MissingRegulatoryElementsForCrosswalksValidator::operator()(
+  const lanelet::LaneletMap & map)
+{
+  lanelet::validation::Issues issues;
+
+  lanelet::autoware::validation::appendIssues(
+    issues, checkMissingRegulatoryElementsForCrosswalks(map));
+
+  return issues;
+}
+
+lanelet::validation::Issues
+MissingRegulatoryElementsForCrosswalksValidator::checkMissingRegulatoryElementsForCrosswalks(
+  const lanelet::LaneletMap & map)
+{
+  lanelet::validation::Issues issues;
+
+  // Get all lanelets whose type is crosswalk
+  std::set<lanelet::Id> cw_ids;
+  std::set<lanelet::Id> tl_elem_with_cw_;
+  for (const auto & ll : map.laneletLayer) {
+    const auto & attrs = ll.attributes();
+    const auto & it = attrs.find(lanelet::AttributeName::Subtype);
+    // Check if this lanelet is crosswalk
+    if (it != attrs.end() && it->second == lanelet::AttributeValueString::Crosswalk) {
+      cw_ids.insert(ll.id());
+
+      // Check if crosswalk has reg elem of traffic light
+      for (const auto & elem : ll.regulatoryElements()) {
+        const auto & attrs = elem->attributes();
+        const auto & it = attrs.find(lanelet::AttributeName::Subtype);
+        if (it != attrs.end() && it->second == lanelet::AttributeValueString::TrafficLight) {
+          tl_elem_with_cw_.insert(elem->id());
+        }
+      }
```

Comment:
It seems that the result of this process is not being used anywhere.

Reply:
Thanks, you are right.
I fixed in here. https://github.com/autowarefoundation/autoware_tools/pull/118/commits/c2a437cd2286b4485b86d770a48548541d1149c8